src/test/java/com/example/batteryprice/repository/BatteryRepositoryTest.java

```diff
@@ -0,0 +1,36 @@
+package com.example.batteryprice.repository;
+
+import com.example.batteryprice.model.Battery;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.data.mongo.DataMongoTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@DataMongoTest
+public class BatteryRepositoryTest {
```

Comment:
I'm curious, what happens in MongoDB here? Since you don't have neither MongoTemplate, nor MongoRepository mocked, I assume that data is written directly in the database. MongoTemplate can be mocked, since it is an attribute in the repository class, but MongoRepository can't. There is a solution for that also, take a look at Mockito's spies which will help you mock methods of a real object.

Reply:
@daliborjovanovic What's the result for this?